libbeat/cmd/instance/metrics.go

```diff
@@ -55,13 +59,29 @@ func setupMetrics(name string) error {
 		IncludeTop:   process.IncludeTopConfig{},
 	}
 	err := beatProcessStats.Init()
+	if err != nil {
+		return err
+	}
+
+	beatProcessSysInfo, err = sysinfo.Self()
+	if err != nil {
+		return err
+	}
 
-	return err
+	var ok bool
+	handleCounter, ok = beatProcessSysInfo.(types.OpenHandleCounter)
+	if !ok {
+		return fmt.Errorf("cannot convert process to OpenHandleCounter: %v", beatProcessSysInfo)
```

Comment:
Stating that the process does not implement OpenHandleCounter would be closer to the wording used by the language specification for a [type assertion](https://golang.org/ref/spec#Type_assertions).

But I imagine this error will be removed (or replaced by a log statement).